src/features/bivariate_manager/components/BivariateGreetings/BivariateGreetingsContainer.tsx

```diff
@@ -0,0 +1,52 @@
+import { Close24 } from '@konturio/default-icons';
+import cn from 'clsx';
+import { Suspense, useEffect, useState } from 'react';
+import { lazily } from 'react-lazily';
+import { GREETINGS_DISABLED_COOKIE } from '~features/bivariate_manager/constants';
+import style from './BivariateGreetingsContainer.module.css';
+
+interface BivariateGreetingsContainerProps {
+  className?: string;
+}
+
+function getCookie(name: string) {
+  const nameEQ = name + '=';
+  const ca = document.cookie.split(';');
+  for (let i = 0; i < ca.length; i++) {
+    let c = ca[i];
+    while (c.charAt(0) == ' ') c = c.substring(1, c.length);
+    if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length, c.length);
+  }
+  return null;
+}
+
+const { BivariateGreetings } = lazily(() => import('./BivariateGreetings'));
+
+export const BivariateGreetingsContainer = ({
+  className,
+}: BivariateGreetingsContainerProps) => {
+  const [isComponentShown, setComponentShown] = useState<boolean>(false);
+
+  useEffect(() => {
+    const disabledCookie = getCookie(GREETINGS_DISABLED_COOKIE);
+    if (!disabledCookie && !isComponentShown) {
+      setComponentShown(true);
+    }
+  });
+
+  const onCloseBtnClick = () => {
+    setComponentShown(false);
```

Comment:
It's unintuitive to read for me. Why if user pushes the close button component set as not shown? 
As i understand here's the dialog window that can start showing the component. So if close button pushed, we just don't change default state - component wasn't shown, no?

Reply:
Internal state hides the whole **BivariateGreetingsContainer** component not just close button. So the logic here is: user preseses close btn -> internal component state gets chose to "not shown"  -> component hides itstelf (starts to ouput null instead of greetings element)